### PR TITLE
feat: Add support for initialization, shutdown, and provider status events.

### PR DESCRIPTION
**Pull request by kinyoklion**
I will add configuration change event support in another PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,23 +29,26 @@ This version of the SDK is built for the following targets:
...
 
-var ldClient  = new LdClient(config);
-var provider = new Provider(ldClient);
+var provider = new Provider(config);
+
+// If you need access to the LdClient, then you can use GetClient().
+// This can be used for use-cases that are not supported by OpenFeature such as migration flags and track events.
+var ldClient = provider.GetClient()
 
 OpenFeature.Api.Instance.SetProvider(provider);
```
Comment (kinyoklion) [inline README.md:53]: I need to change these to use SetProviderAsync.

File: src/LaunchDarkly.OpenFeature.ServerProvider/Provider.StatusProvider.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                {
+                    Type = type,
+                    ProviderName = _providerName
+                };
+                if (message != null)
+                {
+                    payload.Message = message;
+                }
+
+                if (!_eventChannel.Writer.TryWrite(payload))
```
Comment (kinyoklion) [inline src/LaunchDarkly.OpenFeature.ServerProvider/Provider.StatusProvider.cs:37]: Apparently this is bounded to 1, so I have to use the async version, which is going to make this chaotic.

**Review by tanderson-ld — APPROVED** [review]

File: src/LaunchDarkly.OpenFeature.ServerProvider/Provider.cs [hunk 3]
```diff
@@ -84,6 +113,72 @@ public Provider(ILdClient client, ProviderConfiguration config = null)
...
+            }
+
+            return _initCompletion.Task;
+        }
+
+        /// <inheritdoc />
+        public override Task Shutdown()
+        {
+            _client.DataSourceStatusProvider.StatusChanged -= StatusChangeHandler;
+            (_client as IDisposable)?.Dispose();
```
Comment (tanderson-ld) [inline src/LaunchDarkly.OpenFeature.ServerProvider/Provider.cs:155]: Is this cast required?

Comment (kinyoklion) [reply src/LaunchDarkly.OpenFeature.ServerProvider/Provider.cs:155]: Yes. The `LdClient` is disposable, it implements ILdClient and IDisposable. ILdClient doesn't inherit from disposable.

This follows the interface segregation principle. Typically the idea is that an implementation of some specific interface doesn't mean that the implementation also implements other interfaces.

We may or may not want to consider if that is the correct choice, but it is reasonably idiomatic.

Comment (tanderson-ld) [reply src/LaunchDarkly.OpenFeature.ServerProvider/Provider.cs:155]: I see it now, I thought _client's type was the concrete type.

**Pull request merged**